icicle/utils/error_handler.cuh

```diff
@@ -34,4 +36,27 @@ void inline syncDevice(const char* const file, const int line)
   }
 }
 
+#define CHECK_LAST_IS_STICKY_ERROR() checkLastStickyError(__FILE__, __LINE__)
+cudaError_t inline checkLastStickyError(const char* const file, const int line)
+{
+  cudaError_t err{cudaGetLastError()};
+  if (err != cudaSuccess) {
+    std::cerr << "CUDA Runtime Error at: " << file << ":" << line << std::endl;
+    std::cerr << cudaGetErrorString(err) << std::endl;
+    // check for sticky (unrecoverable) error where context is corrupted
+    // and the only option is to restart process
+    err = cudaDeviceSynchronize(); // TODO: only one call to cudaDeviceSynchronize seems sufficent according to
```

Comment:
This can be a pretty heavy operation. Is calling `cudaGetLastError()` twice in a row and seeing both have the same error enough for detecting stickiness since the first one won't reset the error if it is sticky

Reply:
if I understand you correctly - according to[ this thread](https://forums.developer.nvidia.com/t/cuda-errors-determine-sticky-ness/271625) - and comment by [Robert Crovella](https://developer.nvidia.com/blog/author/bob-crovella/)  
_It seems sufficient simply to **call `cudaDeviceSynchronize()` twice** (or even just once). If the same error (or any error) is indicated, its a pretty good indicator_  . 

Just calling `cudaGetLastError()` won't be enough to determine stickiness - the first call resets the error. In the current draft - the rationale is if we have some error, it's worth checking "stickiness" at the expense of  `cudaDeviceSynchronize()` - but once, assuming it's sufficient. Anyway, this will be tested once we settle on the approach.

Reply:
> Just calling cudaGetLastError() twice won't be enough to determine stickiness - the first call resets the error.

Robert had an answer [here](https://stackoverflow.com/questions/43659314/how-can-i-reset-the-cuda-error-to-success-with-driver-api-after-a-trap-instructi) that says calling `cudaGetLastError()` **will not reset** a sticky error. I don't see why that isn't a sufficient test for stickiness